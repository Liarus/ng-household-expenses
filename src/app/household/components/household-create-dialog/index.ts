export * from './household-create-dialog.component';
