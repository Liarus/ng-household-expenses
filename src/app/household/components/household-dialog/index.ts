export * from './household-dialog.component';
