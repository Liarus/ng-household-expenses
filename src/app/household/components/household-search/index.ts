export * from './household-search.component';
