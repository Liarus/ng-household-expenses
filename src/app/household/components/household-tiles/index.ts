export * from './household-tiles.component';
