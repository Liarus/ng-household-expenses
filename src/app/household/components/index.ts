export * from './household-list';
export * from './household-dialog';
export * from './household-tiles';
export * from './household-search';
