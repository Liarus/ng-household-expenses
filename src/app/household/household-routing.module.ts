import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

import { HouseholdPageComponent } from './containers';

const routes: Routes = [{ path: '', component: HouseholdPageComponent }];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class HouseholdRoutingModule {}
