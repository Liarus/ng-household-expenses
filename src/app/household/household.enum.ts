export enum HouseholdSorting {
  NameAsc = 1,
  NameDesc = 2,
  SymbolAsc = 3,
  SymbolDesc = 4,
  DescriptionAsc = 5,
  DescriptionDesc = 6
}
