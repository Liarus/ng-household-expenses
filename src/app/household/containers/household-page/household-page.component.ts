import { Component, OnInit, ChangeDetectionStrategy, OnDestroy } from '@angular/core';
import { Store, select } from '@ngrx/store';
import { takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';

import * as fromAuth from '../../../auth/store/reducers';
import * as fromRoot from '../../../store/reducers';
import * as fromCore from '../../../core/store/reducers';
import * as fromHousehold from '../../store/reducers';
import * as HouseholdActions from '../../store/actions/household.actions';
import { User } from 'src/app/auth/models/user.model';


@Component({
  selector: 'app-household-page',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
    <div class="app-basic-container">
      <app-household-list [isLoading]='isLoading$ | async'
        [isMobile]='isMobile$ | async'
        [households]='household$ | async'
        (create)=onCreate()
        (remove)=onRemove($event)
        (edit)=onEdit($event)
      ></app-household-list>
    </div>
  `
})
export class HouseholdPageComponent implements OnInit, OnDestroy {
  loggedUser$ = this.store.pipe(select(fromAuth.getLoggedUser));
  isLoading$ = this.store.pipe(select(fromHousehold.getHouseholdsLoading));
  isMobile$ = this.store.pipe(select(fromCore.getIsMobile));
  household$ = this.store.pipe(select(fromHousehold.getAllHouseholds));

  private unsubscribe: Subject<void> = new Subject();
  private userId: string;

  constructor(private store: Store<fromRoot.State>) {
    this.loggedUser$
      .pipe(takeUntil(this.unsubscribe))
      .subscribe((loggedIn: User) => {
        if (loggedIn) {
          this.userId = loggedIn.id;
          this.store.dispatch(new HouseholdActions.LoadHouseholds({userId: this.userId}));
        }
      });
  }

  ngOnInit() {
  }

  ngOnDestroy(): void {
    this.unsubscribe.next();
    this.unsubscribe.complete();
  }

  onCreate() {
    this.store.dispatch(new HouseholdActions.OpenCreateHouseholdDialog({userId: this.userId}));
  }

  onRemove(id: string) {
    this.store.dispatch(new HouseholdActions.RemoveHousehold({householdId: id}));
  }

  onEdit(id: string) {
    this.store.dispatch(new HouseholdActions.OpenEditHouseholdDialog({userId: this.userId, householdId: id}));
  }
}
