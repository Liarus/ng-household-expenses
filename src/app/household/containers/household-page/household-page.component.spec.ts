/// <reference types="jest" />
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { StoreModule, Store } from '@ngrx/store';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { HAMMER_LOADER } from '@angular/platform-browser';
import { ChangeDetectionStrategy } from '@angular/core';

import { HouseholdPageComponent } from './household-page.component';
import { MaterialModule } from '../../../material/material.module';
import { MockStore } from '../../../shared/tests/mockStore';
import { HouseholdListComponent, HouseholdTilesComponent } from '../../components';
import { User } from '../../../auth/models/user.model';
import * as HouseholdActions from '../../store/actions/household.actions';
import { HouseholdFilter } from '../../models/householdFilter.model';

describe('HouseholdPageComponent', () => {
  let component: HouseholdPageComponent;
  let fixture: ComponentFixture<HouseholdPageComponent>;

  let store: MockStore<any>;
  const user: User = {
    id: '7bb78f33-0612-409e-a1d6-4341fcee9a7e',
    name: 'UserName'
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        MaterialModule,
        NoopAnimationsModule,
        StoreModule.forRoot({})
      ],
      declarations: [
        HouseholdPageComponent,
        HouseholdListComponent,
        HouseholdTilesComponent
      ],
      providers: [
        { provide: Store, useClass: MockStore },
        { provide: HAMMER_LOADER, useValue: () => new Promise(() => {}) }
      ]
    })
    .overrideComponent(HouseholdPageComponent, {
      set: { changeDetection: ChangeDetectionStrategy.Default }
    })
    .compileComponents();
  }));

  beforeEach(() => {
    store = TestBed.get(Store);
    store.setState({
      layout: {
        isMobile: false
      },
      auth: {
        status: {
          user: user
        }
      },
      households: {
        loading: false,
        ids: [
          '550416ea-b523-4468-ae10-ea07d35eb9f0',
          '55798c3b-5551-489b-9dd2-d7e59691a368',
          'b28e143a-a64a-469a-9704-a294cc7356cf'
        ],
        entities: {
          '550416ea-b523-4468-ae10-ea07d35eb9f0': {
            id: '550416ea-b523-4468-ae10-ea07d35eb9f0',
            name: 'Household1 Name',
            symbol: 'Household1 symbol',
            description: 'Household1 description',
            street: 'Household1 street',
            city: 'Household1 city',
            country: 'Household1 country',
            zipCode: 'Household1 zipCode',
            version: 1
          },
          '55798c3b-5551-489b-9dd2-d7e59691a368': {
            id: '55798c3b-5551-489b-9dd2-d7e59691a368',
            name: 'Household2 Name',
            symbol: 'Household2 symbol',
            description: 'Household2 description',
            street: 'Household2 street',
            city: 'Household2 city',
            country: 'Household2 country',
            zipCode: 'Household2 zipCode',
            version: 1
          },
          'b28e143a-a64a-469a-9704-a294cc7356cf': {
            id: 'b28e143a-a64a-469a-9704-a294cc7356cf',
            name: 'Household3 Name',
            symbol: 'Household3 symbol',
            description: 'Household3 description',
            street: 'Household3 street',
            city: 'Household3 city',
            country: 'Household3 country',
            zipCode: 'Household3 zipCode',
            version: 1
          }
        }
      }
    });
    fixture = TestBed.createComponent(HouseholdPageComponent);
    component = fixture.componentInstance;
    spyOn(store, 'dispatch');
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch OpenCreateHouseholdDialog action', () => {
    const expected = new HouseholdActions.OpenCreateHouseholdDialog({
      userId: user.id
    });

    component.onCreate();

    expect(store.dispatch).toHaveBeenLastCalledWith(expected);
  });

  it('should dispatch OpenEditHouseholdDialog action', () => {
    const householdId = '550416ea-b523-4468-ae10-ea07d35eb9f0';
    const expected = new HouseholdActions.OpenEditHouseholdDialog({
      userId: user.id,
      householdId: householdId
    });

    component.onEdit(householdId);

    expect(store.dispatch).toHaveBeenCalledWith(expected);
  });

  it('should dispatch RemoveHousehold action', () => {
    const householdId = '550416ea-b523-4468-ae10-ea07d35eb9f0';
    const expected = new HouseholdActions.RemoveHousehold({
      householdId: householdId
    });

    component.onRemove(householdId);

    expect(store.dispatch).toHaveBeenCalledWith(expected);
  });

  it('should dispatch ApplyFilter action', () => {
    const filterChange = {
      searchText: 'Household1'
    } as Partial<HouseholdFilter>;
    const expected = new HouseholdActions.ApplyFilter(filterChange);

    component.onFilterChanged(filterChange);

    expect(store.dispatch).toHaveBeenCalledWith(expected);
  });

  it('should dispatch InitHouseholds action on isMobile change', () => {
    const expected = new HouseholdActions.InitHouseholds();
    store.setState({
      layout: {
        isMobile: true
      },
      auth: {
        status: {
          user: Object.assign({}, user)
        }
      },
      households: {
        households: {
          ids: [],
        }
      }
    });

    fixture.detectChanges();
    expect(store.dispatch).toHaveBeenCalledWith(expected);
  });
});
