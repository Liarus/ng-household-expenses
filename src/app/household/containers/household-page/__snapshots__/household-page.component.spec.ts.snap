// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseholdPageComponent should match snapshot 1`] = `
<app-household-page
  household$={[Function Store]}
  isLoading$={[Function Store]}
  isMobile$={[Function Store]}
  loggedUser$={[Function Store]}
  store={[Function MockStore]}
  unsubscribe={[Function Subject]}
  userId={[Function String]}
>
  <div
    class="app-basic-container"
  >
    <app-household-list
      ng-reflect-households="[object Object],[object Object"
      ng-reflect-is-loading="false"
      ng-reflect-is-mobile="false"
    >
      <div
        fxlayout="row"
        fxlayoutalign="start center"
        fxlayoutgap="10px"
      >
        <mat-form-field
          class="mat-form-field ng-tns-c7-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder _mat-animation-noopable"
          fxflex=""
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored"
                  id="mat-input-1"
                  matinput=""
                  ng-reflect-placeholder="Filter"
                  placeholder="Filter"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                  <label
                    aria-owns="mat-input-1"
                    class="mat-form-field-label ng-tns-c7-0 mat-empty mat-form-field-empty ng-star-inserted"
                    for="mat-input-1"
                    id="mat-form-field-label-3"
                    ng-reflect-disabled="true"
                    ng-reflect-ng-switch="false"
                  >
                    
                    
                    Filter
                    
                    
                  </label>
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-underline ng-tns-c7-0 ng-star-inserted"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-form-field-hint-wrapper ng-tns-c7-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <button
          class="mat-icon-button _mat-animation-noopable"
          id="household-btn-add"
          mat-icon-button=""
        >
          <span
            class="mat-button-wrapper"
          >
            <mat-icon
              aria-hidden="true"
              class="mat-icon material-icons"
              role="img"
            >
              add
            </mat-icon>
          </span>
          <div
            class="mat-button-ripple mat-ripple mat-button-ripple-round"
            matripple=""
            ng-reflect-centered="true"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
      </div>
      <mat-table
        class="mat-elevation-z8 mat-table"
        matsort=""
        ng-reflect-data-source="[object Object]"
        role="grid"
      >
        
        <mat-header-row
          class="mat-header-row ng-star-inserted"
          ng-reflect-ng-class="[object Object]"
          role="row"
        >
          
          <mat-header-cell
            class="mat-header-cell ng-tns-c11-1 cdk-column-name mat-column-name ng-star-inserted"
            mat-sort-header=""
            ng-reflect-id=""
            role="columnheader"
          >
            <div
              class="mat-sort-header-container"
            >
              <button
                aria-label="Change sorting for name"
                class="mat-sort-header-button"
                type="button"
              >
                 Name 
              </button>
              <div
                class="mat-sort-header-arrow ng-trigger ng-trigger-arrowPosition"
                style="transform:translateY(25%);opacity:0;0:transform;1:opacity;"
              >
                <div
                  class="mat-sort-header-stem"
                />
                <div
                  class="mat-sort-header-indicator ng-trigger ng-trigger-indicator"
                  style="transform:translateY(0px);0:transform;"
                >
                  <div
                    class="mat-sort-header-pointer-left ng-trigger ng-trigger-leftPointer"
                    style="transform:rotate(-45deg);0:transform;"
                  />
                  <div
                    class="mat-sort-header-pointer-right ng-trigger ng-trigger-rightPointer"
                    style="transform:rotate(45deg);0:transform;"
                  />
                  <div
                    class="mat-sort-header-pointer-middle"
                  />
                </div>
              </div>
            </div>
          </mat-header-cell>
          <mat-header-cell
            class="mat-header-cell ng-tns-c11-2 cdk-column-symbol mat-column-symbol ng-star-inserted"
            mat-sort-header=""
            ng-reflect-id=""
            role="columnheader"
          >
            <div
              class="mat-sort-header-container"
            >
              <button
                aria-label="Change sorting for symbol"
                class="mat-sort-header-button"
                type="button"
              >
                 Symbol 
              </button>
              <div
                class="mat-sort-header-arrow ng-trigger ng-trigger-arrowPosition"
                style="transform:translateY(25%);opacity:0;0:transform;1:opacity;"
              >
                <div
                  class="mat-sort-header-stem"
                />
                <div
                  class="mat-sort-header-indicator ng-trigger ng-trigger-indicator"
                  style="transform:translateY(0px);0:transform;"
                >
                  <div
                    class="mat-sort-header-pointer-left ng-trigger ng-trigger-leftPointer"
                    style="transform:rotate(-45deg);0:transform;"
                  />
                  <div
                    class="mat-sort-header-pointer-right ng-trigger ng-trigger-rightPointer"
                    style="transform:rotate(45deg);0:transform;"
                  />
                  <div
                    class="mat-sort-header-pointer-middle"
                  />
                </div>
              </div>
            </div>
          </mat-header-cell>
          <mat-header-cell
            class="mat-header-cell ng-tns-c11-3 cdk-column-description mat-column-description ng-star-inserted"
            mat-sort-header=""
            ng-reflect-id=""
            role="columnheader"
          >
            <div
              class="mat-sort-header-container"
            >
              <button
                aria-label="Change sorting for description"
                class="mat-sort-header-button"
                type="button"
              >
                 Description 
              </button>
              <div
                class="mat-sort-header-arrow ng-trigger ng-trigger-arrowPosition"
                style="transform:translateY(25%);opacity:0;0:transform;1:opacity;"
              >
                <div
                  class="mat-sort-header-stem"
                />
                <div
                  class="mat-sort-header-indicator ng-trigger ng-trigger-indicator"
                  style="transform:translateY(0px);0:transform;"
                >
                  <div
                    class="mat-sort-header-pointer-left ng-trigger ng-trigger-leftPointer"
                    style="transform:rotate(-45deg);0:transform;"
                  />
                  <div
                    class="mat-sort-header-pointer-right ng-trigger ng-trigger-rightPointer"
                    style="transform:rotate(45deg);0:transform;"
                  />
                  <div
                    class="mat-sort-header-pointer-middle"
                  />
                </div>
              </div>
            </div>
          </mat-header-cell>
          <mat-header-cell
            class="mat-header-cell cdk-column-actions mat-column-actions ng-star-inserted"
            role="columnheader"
          >
             Actions 
          </mat-header-cell>
        </mat-header-row>
        
        <mat-row
          class="mat-row ng-star-inserted"
          ng-reflect-ng-class="[object Object]"
          role="row"
        >
          
          <mat-cell
            class="mat-cell cdk-column-name mat-column-name ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Name:
            </span>
             Household1 Name 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-symbol mat-column-symbol ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Symbol:
            </span>
             Household1 symbol 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-description mat-column-description ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Description:
            </span>
             Household1 description 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-actions mat-column-actions ng-star-inserted"
            fxlayout="row"
            fxlayoutgap="20px"
            role="gridcell"
          >
            <button
              class="mat-raised-button _mat-animation-noopable"
              mat-raised-button=""
            >
              <span
                class="mat-button-wrapper"
              >
                Edit
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
            <button
              class="mat-raised-button mat-warn _mat-animation-noopable"
              color="warn"
              mat-raised-button=""
              ng-reflect-color="warn"
            >
              <span
                class="mat-button-wrapper"
              >
                Remove
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </mat-cell>
        </mat-row>
        <mat-row
          class="mat-row ng-star-inserted"
          ng-reflect-ng-class="[object Object]"
          role="row"
        >
          
          <mat-cell
            class="mat-cell cdk-column-name mat-column-name ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Name:
            </span>
             Household2 Name 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-symbol mat-column-symbol ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Symbol:
            </span>
             Household2 symbol 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-description mat-column-description ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Description:
            </span>
             Household2 description 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-actions mat-column-actions ng-star-inserted"
            fxlayout="row"
            fxlayoutgap="20px"
            role="gridcell"
          >
            <button
              class="mat-raised-button _mat-animation-noopable"
              mat-raised-button=""
            >
              <span
                class="mat-button-wrapper"
              >
                Edit
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
            <button
              class="mat-raised-button mat-warn _mat-animation-noopable"
              color="warn"
              mat-raised-button=""
              ng-reflect-color="warn"
            >
              <span
                class="mat-button-wrapper"
              >
                Remove
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </mat-cell>
        </mat-row>
        <mat-row
          class="mat-row ng-star-inserted"
          ng-reflect-ng-class="[object Object]"
          role="row"
        >
          
          <mat-cell
            class="mat-cell cdk-column-name mat-column-name ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Name:
            </span>
             Household3 Name 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-symbol mat-column-symbol ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Symbol:
            </span>
             Household3 symbol 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-description mat-column-description ng-star-inserted"
            ng-reflect-ng-class="[object Object]"
            role="gridcell"
          >
            <span
              class="app-table-mobile-label app-none"
              ng-reflect-klass="app-table-mobile-label"
              ng-reflect-ng-class="[object Object]"
            >
              Description:
            </span>
             Household3 description 
          </mat-cell>
          <mat-cell
            class="mat-cell cdk-column-actions mat-column-actions ng-star-inserted"
            fxlayout="row"
            fxlayoutgap="20px"
            role="gridcell"
          >
            <button
              class="mat-raised-button _mat-animation-noopable"
              mat-raised-button=""
            >
              <span
                class="mat-button-wrapper"
              >
                Edit
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
            <button
              class="mat-raised-button mat-warn _mat-animation-noopable"
              color="warn"
              mat-raised-button=""
              ng-reflect-color="warn"
            >
              <span
                class="mat-button-wrapper"
              >
                Remove
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </mat-cell>
        </mat-row>
        
      </mat-table>
      
    </app-household-list>
  </div>
</app-household-page>
`;
